quilt/test/test_push.py

```diff
@@ -37,24 +26,37 @@ def test_push(self):
         pkg_hash = pkg_obj.get_hash()
         assert pkg_hash
         contents = pkg_obj.get_contents()
-        urls = upload_urls(contents)
-        for url in urls.values():
-            self._mock_s3(url)
 
-        self._mock_put_package('foo/bar', pkg_hash, contents)
+        all_hashes = set(find_object_hashes(contents))
+        for blob_hash in all_hashes:
+            head_url = "https://example.com/head/{owner}/{hash}".format(
+                owner='foo', hash=blob_hash)
+            put_url = "https://example.com/put/{owner}/{hash}".format(
+                owner='foo', hash=blob_hash)
+            self._mock_get_blob('foo', blob_hash, head_url, put_url)
+            self._mock_s3(head_url, put_url)
+
+        self._mock_put_package('foo/bar', pkg_hash)
```

Comment:
so this is just an end-to-end test of pushing a package? just wondering if it's specific to this PR. given that we now dedupe, it's also slightly suspicous that the test guarantees unique hashes with `set`. since `set` is part of the `command.py.push` logic, wouldn't it be better to rely on its presence there?